hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -2391,42 +2406,91 @@ public void run(Timeout timeout) throws Exception {
       return failedFuture(new IllegalArgumentException("Passed region name can't be null"));
     }
     try {
+      TableName parentTable;
       CompletableFuture<Optional<HRegionLocation>> future;
       if (RegionInfo.isEncodedRegionName(regionNameOrEncodedRegionName)) {
         String encodedName = Bytes.toString(regionNameOrEncodedRegionName);
-        if (encodedName.length() < RegionInfo.MD5_HEX_LENGTH) {
-          // old format encodedName, should be meta region
+
+        //TODO francis do we really need to support encoded name for root?
```

Comment:
Dunno. I think the region id used to be 0. Could it be the encoded name too?

The encoded name is usually md5 hash but old style Regions like hbase:meta used the original simple hash encoded number.

Reply:
Oh this question was whether we wanted to support the encoded name or root as argument for "regionNameOrEncodedRegionName"? Becuase then I have to do the succeeding not so nice thing of enumerating all the root replicas and check if there is a match.